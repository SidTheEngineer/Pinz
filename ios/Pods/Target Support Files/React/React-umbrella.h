#ifdef __OBJC__
#import <UIKit/UIKit.h>
#endif

#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTWebSocketProxy.h"
#import "RCTWebSocketProxyDelegate.h"
#import "CSSLayout.h"
#import "CSSMacros.h"
#import "CSSNodeList.h"
#import "RCTJSCErrorHandling.h"
#import "RCTJSCExecutor.h"
#import "RCTJSCWrapper.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTRedBox.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTFPSGraph.h"
#import "RCTJSCProfiler.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+MapKit.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTMap.h"
#import "RCTMapAnnotation.h"
#import "RCTMapManager.h"
#import "RCTMapOverlay.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTNavigator.h"
#import "RCTNavigatorManager.h"
#import "RCTNavItem.h"
#import "RCTNavItemManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTabBar.h"
#import "RCTTabBarItem.h"
#import "RCTTabBarItemManager.h"
#import "RCTTabBarManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewControllerProtocol.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTActionSheetManager.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTRawTextManager.h"
#import "RCTShadowRawText.h"
#import "RCTShadowText.h"
#import "RCTText.h"
#import "RCTTextField.h"
#import "RCTTextFieldManager.h"
#import "RCTTextManager.h"
#import "RCTTextSelection.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTVibration.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketManager.h"
#import "RCTWebSocketModule.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

