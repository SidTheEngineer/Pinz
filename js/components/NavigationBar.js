import React from 'react';
import {
  Text,
  Navigator,
  StyleSheet,
  TouchableHighlight
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const styles = StyleSheet.create({
  navigationBar: {
    backgroundColor: 'rgb(51, 204, 94)'
  },
  iconContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  routeTitle: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    color: 'white'
  }
});

const NavigationBar = (
  <Navigator.NavigationBar
    style={styles.navigationBar}
    navigationStyles={Navigator.NavigationBar.StylesIOS}
    routeMapper={{
      LeftButton: () => (
        <TouchableHighlight style={[styles.iconContainer, { marginLeft: 20 }]}>
          <Icon name="arrow-left" size={20} color={'white'} />
        </TouchableHighlight>
      ),
      RightButton: () => (
        <TouchableHighlight style={[styles.iconContainer, { marginRight: 20 }]}>
          <Icon name="arrow-right" size={20} color={'white'} />
        </TouchableHighlight>
      ),
      Title: route => (<Text style={styles.routeTitle}>{ route.title }</Text>)
    }}
  />
);

export default NavigationBar;
