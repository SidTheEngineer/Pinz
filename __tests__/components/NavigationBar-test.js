import { Navigator } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import NavigationBar from '../../js/components/NavigationBar';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');

describe('NavigationBar', () => {
  it('renders correctly', () => {
    const navigationBar = renderer.create(
      <Navigator
        navigationBar={NavigationBar}
        renderScene={jest.fn()}
        initialRoute={{ title: 'test' }}
      />
    ).toJSON();
    expect(navigationBar).toMatchSnapshot();
  });
});
