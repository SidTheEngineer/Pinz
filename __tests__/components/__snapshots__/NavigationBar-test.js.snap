exports[`NavigationBar renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }>
    <View
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          null,
        ]
      } />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(79,232,122)",
          "height": 64,
          "justifyContent": "center",
        },
      ]
    }>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": 44,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 6,
          }
        }>
        test
      </Text>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 44,
          "left": 0,
          "opacity": 1,
          "overflow": "hidden",
          "position": "absolute",
          "top": 20,
        }
      }>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginTop": 1,
              "width": 80,
            },
            Object {
              "marginLeft": 5,
            },
          ]
        }>
        <Icon
          color="white"
          name="angle-left"
          size={30} />
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
